src/dpl/src/Grid.cpp

```diff
@@ -655,6 +656,16 @@ void Opendp::paintPixel(Cell* cell, int grid_x, int grid_y)
       ++layer_y_end;
     }
 
+    if (layer_y_end >= layer.second.row_count) {
```

Comment:
The equal sign is what is causing the error.
The error message says:
`[ERROR DPL-0049] Cannot paint grid index: 2800 because it is out of bounds. Calculated (row end 6) > (rows 6). This is a bug`

Which is not really a bug since layer_y_end is an exclusive bound. We never touch it as you can see in the loop below.

Reply:
Removing it results in all tests passing.

Reply:
@QuantamHD the ball is in your court

Reply:
@QuantamHD ??

Reply:
Done.